service/discovery/discovery.go

```diff
@@ -100,9 +104,12 @@ const (
 type ServiceOption func(*Service)
 
 // WithAssessmentAddress is an option to configure the assessment service gRPC address.
-func WithAssessmentAddress(address string) ServiceOption {
+func WithAssessmentAddress(address string, opts ...grpc.DialOption) ServiceOption {
 	return func(s *Service) {
-		s.assessmentAddress = address
+		s.assessmentAddress = grpcTarget{
```

Comment:
In the assessment I have done an additional empty string check. I am not sure if it is really necessary: 
`if address == "" {
			address = DefaultAssessmentAddress
		}`

Reply:
Should not be necessary I think